hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineUtils.java

```diff
@@ -246,4 +249,20 @@ public static HoodieCommitMetadata getCommitMetadata(
       return HoodieCommitMetadata.fromBytes(data, HoodieCommitMetadata.class);
     }
   }
+
+  /**
+   * Returns the basic commit timeline. For basic we means 'commit' for COPY_ON_WRITE
+   * and 'delta_commit' for MERGE_ON_READ.
+   */
+  public static HoodieTimeline getBasicCommitTimeline(HoodieTableMetaClient metaClient) {
+    HoodieTableType tableType = metaClient.getTableType();
+    switch (tableType) {
+      case COPY_ON_WRITE:
+        return metaClient.getActiveTimeline().getTimelineOfActions(CollectionUtils.createSet(HoodieTimeline.COMMIT_ACTION));
+      case MERGE_ON_READ:
+        return metaClient.getActiveTimeline().getTimelineOfActions(CollectionUtils.createSet(HoodieTimeline.DELTA_COMMIT_ACTION));
```

Comment:
For MOR table, the compaction will be converted to commit_action after finished, here we exclude COMMIT_ACTION intentionally ?

Reply:
We should filter pending instants for the caller.